tests/Unit/DataStructures/Test_ModalVector.cpp

```diff
@@ -0,0 +1,99 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <array>
+#include <tuple>
+
+#include "DataStructures/ModalVector.hpp"    // IWYU pragma: keep
+#include "ErrorHandling/Error.hpp"
+#include "Utilities/DereferenceWrapper.hpp"  // IWYU pragma: keep
+#include "Utilities/Functional.hpp"
+#include "Utilities/StdHelpers.hpp"  // IWYU pragma: keep
+#include "Utilities/TypeTraits.hpp"
+#include "tests/Unit/DataStructures/VectorImplTestHelper.hpp"
+
+// IWYU pragma: no_include "DataStructues/DenseMatrix.hpp"
+// IWYU pragma: no_include "tests/Utilities/MakeWithRandomValues.hpp"
+
+class DataVector;
+
+SPECTRE_TEST_CASE("Unit.DataStructures.ModalVector", "[DataStructures][Unit]") {
```

Comment:
Could you (similarly to DataVector now) combine the `ModalVector` and `ModalVector.Math` tests?